third_party/move/tools/move-bytecode-utils/src/compiled_module_viewer.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::CompiledModule;
+use move_core_types::language_storage::ModuleId;
+use std::borrow::Borrow;
+
+pub trait CompiledModuleView {
+    type Item: Borrow<CompiledModule>;
+
+    fn view_compiled_module(&self, id: &ModuleId) -> anyhow::Result<Self::Item>;
```

Comment:
Note: consider taking addr & module name as separate arguments to allow more clients to use this API

Reply:
I think we should just use addr & module name for this API, should be sufficient. I added a note and will make a pass after this PR lands